pytorch_lightning/loops/utilities.py

```diff
@@ -105,9 +105,11 @@ def _process_training_step_output(
     return results, hiddens
 
 
-def _prepare_dataloader_iter(dataloader_iter: Iterator, batch_idx: int) -> Iterator:
+def _prepare_dataloader_iter(data_fetcher: AbstractDataFetcher, batch_idx: int) -> Iterator:
```

Comment:
How much do we actually need to expose the "AbstractDataFetcher" here versus just an iterator? It seems to me that we're really not adding much value by exposing a new interface here. 

The custom logic below really should be part of the iterator implementation itself as i comment below.

Reply:
yes, the iterator type annotation would have sufficed here